clang/test/CodeGenSYCL/kernel-arg-accessor-pointer.cpp

```diff
@@ -0,0 +1,69 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown -disable-llvm-passes -emit-llvm %s -o - | FileCheck %s
+
+// This test checks if the metadata "kernel-arg-runtime-aligned"
+// is generated if the kernel captures an accessor
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+
+queue q;
+
+int main() {
+
+  using Accessor =
+      accessor<int, 1, access::mode::read_write, access::target::global_buffer>;
+  Accessor acc[2];
```

Comment:
Can please we also include tests for:
* Various types of accessors (e.g. buffers in the `local` address space).
* USM pointers to make sure we're not assuming that those are runtime aligned.

Reply:
I have added test cases for raw pointers vs accessors ( read_write, readOnly). Hope this is sufficcient.
`e.g. accessor in the local address space`
Did you mean [local_accessor](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:accessor.local) ?

Reply:
Yes! You mention [here](https://github.com/intel/llvm/pull/5111/files#diff-e724febedab9c1a2832bf2056d208ff02ddcb2e6f90b5a653afc9b19ac78a5d7R1815) that you're looking for global accessors but you're not really checking the address space. I think the code is correct as is and we should cover local accessors. So, maybe reword the description and the comments to say "all accessors" instead of just "global accessors".